src/semidiscretization/semidiscretization.jl

```diff
@@ -33,7 +33,7 @@ and integrate the result using a quadrature associated with the semidiscretizati
 
 If `normalize` is true, the result is divided by the total volume of the computational domain.
 """
-function integrate(func::Func, u_ode::AbstractVector, semi::AbstractSemidiscretization; normalize=true) where {Func}
+function integrate(func::Func, u_ode::AbstractArray, semi::AbstractSemidiscretization; normalize=true) where {Func}
```

Comment:
Could you please explain the reason for this change?

Reply:
First, thank you very much for looking into this.

We didn't really understand what the purpose of storing `u_ode` as vector and reshaping with `wrap_array` is. Is this only related to AMR?
We tried to store `u_ode` as array, similar to how it is reshaped with `wrap_array`, and we just fed this multi-dimensional array to `OrdinaryDiffEq`, which worked fine. Here, we needed to change this function to accept multi-dimensional arrays.

Looking back now, I think we only chose to change this because we wanted to store `u_ode` with indices `[variable, node_x, node_y, element_x, element_y]` (in 2D) instead of using element IDs, which would need us to rewrite `wrap_array`. We later decided to use linear indexing for the elements again, so we could actually use `wrap_array` again and store `u_ode` as vector, i.e. revert these changes.

Reply:
Yes, exactly. In Julia, multidimensional arrays cannot be `resize!`ed, so we needed to use `Vector`s in the interface to OrdinaryDiffEq for AMR. That's the only reason. (It also simplifies handling of Jacobians a bit, but that's not the main motivation).

Reply:
Okay, we reverted these changes.